specs/phase1/validator.md

```diff
@@ -0,0 +1,520 @@
+
+## Beacon chain responsibilities
+
+A validator has two primary responsibilities to the beacon chain: [proposing blocks](#block-proposal) and [creating attestations](#attestations-1). Proposals happen infrequently, whereas attestations should be created once per epoch.
+
+These responsibilities are largely unchanged from Phase 0, but utilize the updated `SignedBeaconBlock`, `BeaconBlock`,  `BeaconBlockBody`, `Attestation`, and `AttestationData` definitions found in Phase 1. Below notes only the additional and modified behavior with respect to Phase 0.
+
+Phase 1 adds light client committees and associated responsibilities, discussed [below](#light-client-committee).
+
+### Block proposal
+
+#### Preparing for a `BeaconBlock`
+
+`slot`, `proposer_index`, `parent_root` fields are unchanged.
+
+#### Constructing the `BeaconBlockBody`
+
+`randao_reveal`, `eth1_data`, and `graffiti` are unchanged.
+
+`proposer_slashings`, `deposits`, and `voluntary_exits` are unchanged.
+
+`attester_slashings` and `attestations` operate exactly as in Phase 0, but with new definitations of `AttesterSlashing` and `Attestation`, along with modified validation conditions found in `process_attester_slashing` and `process_attestation`.
+
+##### Custody slashings
+
+Up to `MAX_CUSTODY_SLASHINGS`, [`CustodySlashing`](./custody-game.md#custodyslashing) objects can be included in the `block`. The custody slashings must satisfy the verification conditions found in [custody slashings processing](./custody-game.md#custody-slashings). The validator receives a small "whistleblower" reward for each custody slashing included (THIS IS NOT CURRENTLY THE CASE BUT PROBABLY SHOULD BE).
+
+##### Custody key reveals
+
+Up to `MAX_CUSTODY_KEY_REVEALS`, [`CustodyKeyReveal`](./custody-game.md#custodykeyreveal) objects can be included in the `block`. The custody key reveals must satisfy the verification conditions found in [custody key reveal processing](./custody-game.md#custody-key-reveals). The validator receives a small reward for each custody key reveal included.
+
+##### Early derived secret reveals
+
+Up to `MAX_EARLY_DERIVED_SECRET_REVEALS`, [`EarlyDerivedSecretReveal`](./custody-game.md#earlyderivedsecretreveal) objects can be included in the `block`. The early derived secret reveals must satisfy the verification conditions found in [early derived secret reveal processing](./custody-game.md#custody-key-reveals). The validator receives a small "whistleblower" reward for each early derived secrete reveal included.
+
+##### Shard transitions
+
+Exactly `MAX_SHARDS` [`ShardTransition`](./beacon-chain#shardtransition) objects are included in the block. Default each to an empty `ShardTransition()`. Then for each committee assigned to the slot with an associated `committee_index` and `shard`, set `shard_transitions[shard] = full_transitions[winning_root]` if the committee had enough weight to form a crosslink this slot.
```

Comment:
Side question. Should `MAX_SHARDS` really be 1024 as it's currently defined? That's a lot of sparse `ShardTransition()` inside beacon block even at the best case (e.g. 960)

It might be better to align `MAX_SHARDS` to `MAX_COMMITTEES_PER_SLOT`, and increase the shard count in an event of a network upgrade via social consensus

Reply:
Yeah, I see that.. I think the argument for having it in there is that 0s would be almost entirely compressible on the wire (via snappy).

I think it's reasonable to change it.. but we can discuss in another thread